libevmasm/RuleList.h

```diff
@@ -230,58 +239,60 @@ std::vector<SimplificationRule<Pattern>> simplificationRuleListPart5(
 	Pattern
 )
 {
+	using Word = typename Pattern::Word;
+
 	std::vector<SimplificationRule<Pattern>> rules;
 
 	// Replace MOD X, <power-of-two> with AND X, <power-of-two> - 1
-	for (size_t i = 0; i < 256; ++i)
+	for (size_t i = 0; i < Pattern::WordSize; ++i)
 	{
-		u256 value = u256(1) << i;
+		Word value = Word(1) << i;
 		rules.push_back({
-			{Instruction::MOD, {X, value}},
-			[=]() -> Pattern { return {Instruction::AND, {X, value - 1}}; },
+			{Pattern::Builtins::MOD, {X, value}},
+			[=]() -> Pattern { return {Pattern::Builtins::AND, {X, value - 1}}; },
 			false
 		});
 	}
 
 	// Replace SHL >=256, X with 0
 	rules.push_back({
-		{Instruction::SHL, {A, X}},
-		[=]() -> Pattern { return u256(0); },
+		{Pattern::Builtins::SHL, {A, X}},
+		[=]() -> Pattern { return Word(0); },
 		true,
-		[=]() { return A.d() >= 256; }
+		[=]() { return A.d() >= Pattern::WordSize; }
 	});
 
 	// Replace SHR >=256, X with 0
 	rules.push_back({
-		{Instruction::SHR, {A, X}},
-		[=]() -> Pattern { return u256(0); },
+		{Pattern::Builtins::SHR, {A, X}},
+		[=]() -> Pattern { return Word(0); },
 		true,
-		[=]() { return A.d() >= 256; }
+		[=]() { return A.d() >= Pattern::WordSize; }
 	});
 
 	// Replace BYTE(A, X), A >= 32 with 0
 	rules.push_back({
-		{Instruction::BYTE, {A, X}},
-		[=]() -> Pattern { return u256(0); },
+		{Pattern::Builtins::BYTE, {A, X}},
+		[=]() -> Pattern { return Word(0); },
 		true,
-		[=]() { return A.d() >= 32; }
+		[=]() { return A.d() >= Pattern::WordSize / 8; }
 	});
 
 	for (auto const& op: std::vector<Instruction>{
-		Instruction::ADDRESS,
-		Instruction::CALLER,
-		Instruction::ORIGIN,
-		Instruction::COINBASE
+		Pattern::Builtins::ADDRESS,
+		Pattern::Builtins::CALLER,
+		Pattern::Builtins::ORIGIN,
+		Pattern::Builtins::COINBASE
 	})
 	{
-		u256 const mask = (u256(1) << 160) - 1;
+		Word const mask = (Word(1) << 160) - 1;
```

Comment:
Hm - what happens here, if the word size is less than 160? This will probably need to be changed some more before such word sizes are allowed anyways, but maybe putting this whole ``for`` loop in an ``if (Pattern::WordSize > 160)`` now already doesn't hurt?

Reply:
Right, or for now assert that it's not less?

Reply:
Proper solution would probably be to add ``AddressSize`` to ``Pattern`` and to assert ``WordSize >= AddressSize``, ensuring that everything works, if ``WordSize == AddressSize``, but we don't need to do that now.

Reply:
> Right, or for now assert that it's not less?

Yep, that'd be fine, too - I'd assert that it's greater, though - otherwise I'd need to check what boost does with the mask expression when they're equal :-).